readme.md

```diff
@@ -6,18 +6,85 @@
 [!["help wanted" issues](https://img.shields.io/github/issues/xmldom/xmldom/help%20wanted?color=darkgreen&style=flat-square)](https://github.com/xmldom/xmldom/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)
 [![Mutation report](https://camo.githubusercontent.com/ee312c4ebce7784ce9f785757eba5d6e33e6d950/68747470733a2f2f696d672e736869656c64732e696f2f656e64706f696e743f7374796c653d666c61742675726c3d687474707325334125324625324662616467652d6170692e737472796b65722d6d757461746f722e696f2532466769746875622e636f6d25324662726f647962697473253246786d6c646f6d2532466d6173746572)](https://dashboard.stryker-mutator.io/reports/github.com/brodybits/xmldom/master)
 
-A JavaScript implementation of W3C DOM for Node.js, Rhino and the browser. Fully
-compatible with `W3C DOM level2`; and some compatible with `level3`. Supports
-`DOMParser` and `XMLSerializer` interface such as in browser.
+xmldom is a javascript [ponyfill](https://ponyfill.com/) for the following APIs supported in browsers:
+- convert an XML string into a DOM tree (`new DOMParser().parseFromString(xml, mimeType)` => `Document`)
+- create, access and modify a DOM tree (`new DOMImplementation().createDocument(...)` => `Document`, )
+- serialize a DOM tree back into an XML string (`new XMLSerializer().serializeToString(node)` => `string`)
+
+If your code always/only needs to work in a browser, you should rather not use xmldom. But you should (potentially always) use it, when your code needs to also work in other runtimes like NodeJS or Rhino.
+
+[The implementation is based on several specs](docs/architecture.svg):
```

Comment:
I would list the specs here, and *maybe* embed `docs/architecture.svg` in an implementation section near the end. Or should we make a new page that should include implementation details such as embedding `docs/architecture.svg`?

Reply:
The section directly after this sentence show and describes the relation to the different specs. But maybe this link is rather misleading. Maybe it's good enough for now to not link to the achitecture diagram from the readme for now.